pkg/sbom/nodelist.go

```diff
@@ -193,13 +194,13 @@ func (nl *NodeList) MergeEdges(es []*Edge) {
 // More than one root element can be added to the NodeList.
 func (nl *NodeList) AddRootNode(n *Node) {
 	if n.Id == "" {
-		// TODO warn here
-		return
+		logrus.Warnf("Node with empty ID, creating a new Id: %s@%s\n", n.Name, n.Version)
+		n.Id = NewNodeIdentifier(fmt.Sprintf("%s@%s", n.Name, n.Version))
 	}
 
 	for _, id := range nl.RootElements {
 		if id == n.Id {
-			// TODO warn here
+			logrus.Warnf("Node with ID %s already exists in the NodeList, skipping: %s\n", n.Id, n.Name)
```

Comment:
Just return here, i think it's fine if we keep this as a noop. IMHO adding a node as root node should be idempotent.
```suggestion
```